In [195]:
import pandas as pd
import re

In [196]:
trust = pd.read_csv('Desktop/Desktop/Thinkful DS Bootcamp/DataCleaningChallenge/WELLCOME_APCspend2013_forThinkful.csv', encoding='iso-8859-1')

In [197]:
trust.head(10)

PMID/PMCID                                     Publisher  \
0                    NaN                                           CUP   
1             PMC3679557                                           ACS   
2  23043264  PMC3506128                                            ACS   
3    23438330 PMC3646402                                           ACS   
4   23438216 PMC3601604                                            ACS   
5             PMC3579457                                           ACS   
6             PMC3709265                                           ACS   
7   23057412 PMC3495574                                            ACS   
8      PMCID: PMC3780468  ACS (Amercian Chemical Society) Publications   
9      PMCID: PMC3621575  ACS (Amercian Chemical Society) Publications   

                    Journal title  \
0          Psychological Medicine   
1               Biomacromolecules   
2                      J Med Chem   
3                      J Med Chem   
4                      J Org Chem   
5  Journal of Medicinal Chemistry   
6    Journal of Proteome Research   
7                       Mol Pharm   
8            ACS Chemical Biology   
9            ACS Chemical Biology   

                                       Article title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   
5  Comparative Structural and Functional Studies ...   
6  Mapping Proteolytic Processing in the Secretom...   
7  Quantitative silencing of EGFP reporter gene b...   
8  A Novel Allosteric Inhibitor of the Uridine Di...   
9  Chemical proteomic analysis reveals the drugab...   

  COST (£) charged to Wellcome (inc VAT when charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4                                            £685.88   
5                                           £2392.20   
6                                           £2367.95   
7                                            £649.33   
8                                           £1294.59   
9                                           £1294.78

### The challenge is to: 

### -Determine the five most common journals and the total articles for each. 
### -Calculate the mean, median, and standard deviation of the open-access cost per article for each journal

In [198]:
trust.columns

Index(['PMID/PMCID', 'Publisher', 'Journal title', 'Article title',
       'COST (£) charged to Wellcome (inc VAT when charged)'],
      dtype='object')

In [199]:
trust['Journal title'] = trust['Journal title'].str.lower()
trust['Journal title'] = trust['Journal title'].str.capitalize()
trust['Journal title'] = trust['Journal title'].str.strip()

In [200]:
trust['Journal title'].head(40)

0                           Psychological medicine
1                                Biomacromolecules
2                                       J med chem
3                                       J med chem
4                                       J org chem
5                   Journal of medicinal chemistry
6                     Journal of proteome research
7                                        Mol pharm
8                             Acs chemical biology
9                             Acs chemical biology
10    Journal of chemical information and modeling
11                  Journal of medicinal chemistry
12                  Journal of medicinal chemistry
13                                    Biochemistry
14                  Journal of medicinal chemistry
15                                Gastroenterology
16                 Journal of biological chemistry
17                           Journal of immunology
18                           Journal of immunology
19                            A

In [201]:
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [202]:
trust_grouped_title.head()

Journal title
Academy of nutrition and dietetics                               2379.540
Acs chemical biology                                             1418.186
Acs chemical neuroscience                                        1186.800
Acs nano                                                          668.140
Acta crystallographica section d,  biological crystallography     771.420
Name: Cost, dtype: float64

In [204]:
trust.rename(columns={'COST (£) charged to Wellcome (inc VAT when charged)': 'Cost'}, inplace=True)
trust['Cost'] = trust['Cost'].replace("£","")
trust['Cost'].head(10)

0       £0.00
1    £2381.04
2     £642.56
3     £669.64
4     £685.88
5    £2392.20
6    £2367.95
7     £649.33
8    £1294.59
9    £1294.78
Name: Cost, dtype: object

In [205]:
trust['Cost'] = trust['Cost'].str.replace("£", "")
trust['Cost'].head(10)
#adding .str to .replace fixed the issue

0       0.00
1    2381.04
2     642.56
3     669.64
4     685.88
5    2392.20
6    2367.95
7     649.33
8    1294.59
9    1294.78
Name: Cost, dtype: object

In [206]:
trust['Cost'] = trust['Cost'].str.replace("$", "")
trust['Cost'].head(10)

0       0.00
1    2381.04
2     642.56
3     669.64
4     685.88
5    2392.20
6    2367.95
7     649.33
8    1294.59
9    1294.78
Name: Cost, dtype: object

In [207]:
trust['Cost'] = pd.to_numeric(trust['Cost'])

In [208]:
trust = trust[trust.Cost <= 6000 ]

In [209]:
trust = trust[trust.Cost > 0 ]

In [210]:
trust.head()

PMID/PMCID Publisher                   Journal title  \
1             PMC3679557       ACS               Biomacromolecules   
2  23043264  PMC3506128        ACS                      J med chem   
3    23438330 PMC3646402       ACS                      J med chem   
4   23438216 PMC3601604        ACS                      J org chem   
5             PMC3579457       ACS  Journal of medicinal chemistry   

                                       Article title     Cost  
1  Structural characterization of a Model Gram-ne...  2381.04  
2  Fumaroylamino-4,5-epoxymorphinans and related ...   642.56  
3  Orvinols with mixed kappa/mu opioid receptor a...   669.64  
4  Regioselective opening of myo-inositol orthoes...   685.88  
5  Comparative Structural and Functional Studies ...  2392.20

In [211]:
trust['Cost'].describe()

count    2076.000000
mean     1822.933584
std       766.840519
min         9.930000
25%      1268.700000
50%      1851.650000
75%      2302.330000
max      6000.000000
Name: Cost, dtype: float64

In [212]:
trust_grouped_title = trust.groupby(['Journal title'])['Cost'].mean()

In [214]:
trust.groupby(['Journal title']).count().sort_values('Cost', ascending=False)

PMID/PMCID  Publisher  \
Journal title                                                               
Plos one                                                   180        182   
Journal of biological chemistry                             51         52   
Neuroimage                                                  28         29   
Nucleic acids research                                      25         26   
Proceedings of the national academy of sciences             22         22   
Plos pathogens                                              22         22   
Plos genetics                                               22         22   
Plos neglected tropical diseases                            20         20   
Nature communications                                       18         18   
Human molecular genetics                                    16         18   
Brain                                                       11         14   
Movement disorders                                          12         14   
Bmc public health                                           12         14   
Journal of neuroscience                                     10         13   
Developmental cell                                           9         12   
Biochemical journal                                         11         12   
Journal of general virology                                 11         11   
Current biology                                             10         11   
Bmj                                                          5         10   
Malaria journal                                             10         10   
Bmj open                                                     9          9   
Cell reports                                                 9          9   
Development                                                  8          9   
Neuron                                                       8          9   
Journal of virology                                          8          9   
Journal of structural biology                                8          8   
Hepatology                                                   6          8   
Plosone                                                      8          8   
European journal of immunology                               8          8   
Neurobiology of aging                                        8          8   
...                                                        ...        ...   
Hum resour health                                            1          1   
Human brain mapping                                          0          1   
Human mol genetics online                                    1          1   
Human molecular genetics online                              1          1   
Human mutation: variation, informatics, and dis...           1          1   
Human reproduction                                           1          1   
Human resources for health                                   1          1   
Ijtld                                                        0          1   
Health and place                                             1          1   
Health & place                                               0          1   
Health                                                       1          1   
Haematologica/the haematology journal                        1          1   
Frontiers in invertebrate physiology                         1          1   
Frontiers in molecular neuroscience                          0          1   
Frontiers in neural circuits                                 1          1   
Frontiers in neurorobotics                                   1          1   
Frontiers in perception science                              1          1   
Frontiers in schizophrenia                                   1          1   
Frontiers in systems neuroscience                            1          1   
Frontiers in t cell biology                     

### Most common Journal Titles with numbers of articles
1. Plos one - 182
2. Journal of biological chemistry - 52
3. Neuroimage - 29
4. Nucleic acids research - 26

### Three tied for 5th:
Proceedings of the national academy of sciences	- 22
Plos pathogens	- 22
Plos genetics	- 22

In [216]:
trust['Cost'].describe()

count    2076.000000
mean     1822.933584
std       766.840519
min         9.930000
25%      1268.700000
50%      1851.650000
75%      2302.330000
max      6000.000000
Name: Cost, dtype: float64

### Calculate the mean, median, and standard deviation of the open-access cost per article for each journal

Changed the column name to Cost and the summary statistics are all above. 